cassandra-mesos-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/executor/BackupManager.java

```diff
@@ -0,0 +1,127 @@
+package io.mesosphere.mesos.frameworks.cassandra.executor;
+
+import io.mesosphere.mesos.frameworks.cassandra.executor.jmx.JmxConnect;
+import org.apache.cassandra.service.StorageServiceMBean;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+
+public class BackupManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BackupManager.class);
+
+    @NotNull
+    private final JmxConnect jmxConnect;
+
+    @NotNull
+    private final String backupDir;
+
+    public BackupManager(@NotNull final JmxConnect jmxConnect, @NotNull final String backupDir) {
+        this.jmxConnect = jmxConnect;
+        this.backupDir = backupDir;
+    }
+
+    public void backup(@NotNull final String keyspace) throws IOException {
+        backup(keyspace, "s-" + System.currentTimeMillis());
+    }
+
+    public void backup(@NotNull final String keyspace, @NotNull final String snapshot) throws IOException {
+        final StorageServiceMBean storageServiceProxy = jmxConnect.getStorageServiceProxy();
+
+        LOGGER.info("Creating snapshot of keyspace {}", keyspace);
+        storageServiceProxy.takeSnapshot(snapshot, keyspace);
+
+        LOGGER.info("Copying backup of keyspace {}", keyspace);
+        copyKeyspaceSnapshot(snapshot, keyspace);
+
+        LOGGER.info("Clearing snapshot of keyspace {}", keyspace);
+        storageServiceProxy.clearSnapshot(snapshot, keyspace);
+    }
+
+    private void copyKeyspaceSnapshot(final String snapshot, final String keyspace) throws IOException {
+        final List<String> tables = jmxConnect.getColumnFamilyNames(keyspace);
+        for (final String table : tables)
+            copyTableSnapshot(snapshot, keyspace, table);
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
```

Comment:
Used Files.createdDirectories (instead of File.mkdirs()), which implements the above behavior (checks for existence, raise IOException if can't create dir).